llvm/lib/Transforms/IPO/RepoPruning.cpp

```diff
@@ -143,37 +181,25 @@ bool RepoPruning::runOnModule(Module &M) {
     } else {
       auto It = Fragments->find(Repository, Key);
       if (It == Fragments->end(Repository)) {
+        LLVM_DEBUG(dbgs() << "New GO name: " << GO.getName() << '\n');
         InRepository = false;
       } else {
-        // Create  the dependent fragments if existing in the repository.
-        auto Fragment = pstore::repo::fragment::load(Repository, It->second);
-        if (auto Dependents =
-                Fragment->atp<pstore::repo::section_kind::dependent>()) {
-          for (pstore::typed_address<pstore::repo::compilation_member>
-                   Dependent : *Dependents) {
-            auto CM =
-                pstore::repo::compilation_member::load(Repository, Dependent);
-            StringRef MDName = toStringRef(
-                pstore::get_sstring_view(Repository, CM->name).second);
-            auto DMD = TicketNode::get(M.getContext(), MDName,
-                                       toDigestType(CM->digest),
-                                       toGVLinkage(CM->linkage), true);
-            NamedMDNode *const NMD = M.getOrInsertNamedMetadata("repo.tickets");
-            assert(NMD && "NamedMDNode cannot be NULL!");
-            NMD->addOperand(DMD);
-          }
-        }
+        LLVM_DEBUG(dbgs() << "Prunning GO name: " << GO.getName() << '\n');
+        addDependentFragments(M, DependentFragments, Fragments, Repository,
+                              Key);
       }
     }
 
     if (!InRepository) {
       auto It = ModuleFragments.find(Key);
-      // The definition of some global objects may be discarded if not used. If
-      // a global has been pruned and its digest matches a discardable GO, a
-      // missing fragment error might be met during the assembler. To avoid this
-      // issue, this global object can't be pruned if the referenced global
-      // object is discardable.
+      // The definition of some global objects may be discarded if not used.
```

Comment:
It seems there are some formatting inconsistencies. (2 vs 4 spaces).

Reply:
I have verified and formatted this file using the clang-format.